packages/shared/src/hooks/squads/useSquadPostModeration.ts

```diff
@@ -45,51 +48,85 @@ export const rejectReasons: { value: PostModerationReason; label: string }[] = [
   { value: PostModerationReason.Other, label: 'Other' },
 ];
 
-interface UseSquadPostModeration {
-  onApprove: (ids: string[], onSuccess?: MouseEventHandler) => Promise<void>;
-  onReject: (id: string, onSuccess?: MouseEventHandler) => void;
+export interface UseSquadPostModeration {
+  onApprove: (
+    ids: string[],
+    sourceId: string,
+    onSuccess?: MouseEventHandler,
+  ) => Promise<void>;
+  onReject: (
+    id: string,
+    sourceId: string,
+    onSuccess?: MouseEventHandler,
+  ) => void;
   isPending: boolean;
   isSuccess: boolean;
 }
 
-export const useSquadPostModeration = (): UseSquadPostModeration => {
-  const onSuccessRef = useRef<MouseEventHandler>();
-  const { openModal } = useLazyModal();
+export const useSquadPostModeration = ({
+  squad,
+}: {
+  squad: Squad;
+}): UseSquadPostModeration => {
+  const { openModal, closeModal } = useLazyModal();
   const { displayToast } = useToastNotification();
   const { showPrompt } = usePrompt();
-  const onSuccessWrapper = () => {
-    if (onSuccessRef.current) {
-      onSuccessRef.current(null);
-      onSuccessRef.current = undefined;
-    }
+  const { user } = squad.currentMember;
+  const queryClient = useQueryClient();
+  const listQueryKey = generateQueryKey(
+    RequestKey.SquadPostRequests,
+    user,
+    squad.id,
+  );
+  const squadQueryKey = generateQueryKey(RequestKey.Squad, user, squad.handle);
+
+  const invalidateQueries = () => {
+    Promise.all([
+      queryClient.invalidateQueries({ queryKey: listQueryKey }),
+      queryClient.invalidateQueries({ queryKey: squadQueryKey }),
+    ]);
   };
 
   const {
     mutateAsync: onApprove,
     isPending: isPendingApprove,
     isSuccess: isSuccessApprove,
   } = useMutation({
-    mutationFn: (ids: string[]) => squadApproveMutation(ids),
+    mutationFn: ({ postIds, sourceId }: SquadPostModerationProps) =>
+      squadApproveMutation({
+        postIds,
+        sourceId,
+      }),
     onSuccess: () => {
       displayToast('Post(s) approved successfully');
-      onSuccessWrapper();
+      invalidateQueries();
+    },
+    onError: (_, variables) => {
+      if (variables.postIds.length > 50) {
+        displayToast(
+          'Failed to approve post(s). Please approve maximum 50 posts at a time',
+        );
+        return;
+      }
```

Comment:
Any particular reason we have this? Was it a product's requirement?

Reply:
Talked to @capJavert  and he proposed to put the limit on BE + a logger to make sure to catch if this case is a common one. I think this is a good catch considering how rare is to approve 50+ posts at once 👀